dashboard/src/modules/components/TableOfContent/index.jsx

```diff
@@ -1,384 +1,104 @@
 import "./index.less";
 
 import {
-  AngleLeftIcon,
-  DownloadIcon,
-  FolderIcon,
-} from "@patternfly/react-icons";
-import {
-  BadgeToggle,
-  Breadcrumb,
-  BreadcrumbHeading,
-  BreadcrumbItem,
   Divider,
-  DrilldownMenu,
-  Dropdown,
-  DropdownItem,
-  Menu,
-  MenuBreadcrumb,
-  MenuContent,
-  MenuItem,
-  MenuList,
+  Flex,
+  FlexItem,
+  List,
+  ListItem,
+  Sidebar,
+  SidebarContent,
+  SidebarPanel,
 } from "@patternfly/react-core";
-import React, { useEffect, useState } from "react";
-import {
-  TableComposable,
-  Tbody,
-  Td,
-  Th,
-  Thead,
-  Tr,
-} from "@patternfly/react-table";
-import {
-  fetchTOC,
-  updateContentData,
-  updateCurrData,
-  updateSearchSpace,
-  updateStack,
-  updateTableData,
-} from "actions/tableOfContentActions";
+import React, { useEffect } from "react";
+import { fetchTOC, setActiveFileContent } from "actions/tocActions";
 import { useDispatch, useSelector } from "react-redux";
 
-import { DEFAULT_PER_PAGE } from "assets/constants/paginationConstants";
-import { EmptyTable } from "../TableComponent/common-components";
-import { SearchTOC } from "./common-components";
-import TablePagination from "../PaginationComponent";
+import { DownloadIcon } from "@patternfly/react-icons";
+import DrilldownMenu from "./DrillDownMenu";
 import { useParams } from "react-router";
 
 const TableOfContent = () => {
-  const { endpoints } = useSelector((state) => state.apiEndpoint);
-  const [menuDrilledIn, setMenuDrilledIn] = useState([]);
-  const [drilldownPath, setDrillDownPath] = useState([]);
-  const [activeMenu, setActiveMenu] = useState("rootMenu");
-  const [breadCrumb, setBreadCrumb] = useState(null);
-  const [activeFile, setActiveFile] = useState(null);
-  const [breadCrumbLabels, setBreadCrumbLabels] = useState([]);
-  const [param, setParam] = useState("");
-  const [page, setPage] = useState(1);
-  const [perPage, setPerPage] = useState(DEFAULT_PER_PAGE);
-  const params = useParams();
   const dispatch = useDispatch();
-  let dirCount = 0;
-  let fileCount = 0;
+  const params = useParams();
+
+  const { endpoints } = useSelector((state) => state.apiEndpoint);
+  const { activeFile, inventoryDownloadLink } = useSelector(
+    (state) => state.toc
+  );
   useEffect(() => {
     if (Object.keys(endpoints).length > 0)
-      dispatch(fetchTOC(params["dataset_id"], "", false));
+      dispatch(fetchTOC(params["dataset_id"], "", undefined, false));
```

Comment:
It'd be better to use `null` rather than `undefined`, I think. Semantically, Javascript `null` more or less means "intentionally left blank" while `undefined` is more like Javascript's indication that it was "unintentionally left blank". 😆

Reply:
> **`undefined`** is a [primitive](https://developer.mozilla.org/en-US/docs/Glossary/Primitive) value automatically assigned to [...] formal [arguments](https://developer.mozilla.org/en-US/docs/Glossary/Argument) for which there are no actual arguments. ([Source](https://developer.mozilla.org/en-US/docs/Glossary/Undefined))

So, perhaps, in this specific case, `undefined` is reasonable?  🙂

Reply:
The key there is "automatically assigned" ... general recommendation seems to be that the programmer should use `null` and leave `undefined` to Javascript itself. But in this instance, the distinction isn't all that important; and I guess since there are now two approvals, let's just move on.

Reply:
> the programmer should use `null` and leave `undefined` to Javascript itself

That's good to know moving forward.